### PR TITLE
MM-22051: Remove ToJson from model (part 1)

### PR DESCRIPTION
**Pull request by agnivade**
https://mattermost.atlassian.net/browse/MM-22051

```release-note
Removed the following methods/functions:
(ad *AccessData) ToJson()
(ar *AccessResponse) ToJson()
(ar *AnalyticsRow) ToJson()
(ar AnalyticsRows) ToJson()
(o *Audit) ToJson()
(o Audits) ToJson()
(ad *AuthData) ToJson()
(ar *AuthorizeRequest) ToJson()
(o *ChannelPatch) ToJson()
(o *ChannelsWithCount) ToJson()
(o *ChannelCounts) ToJson()
(o *ChannelData) ToJson()
(o *ChannelMembers) ToJson()
(o *ChannelUnread) ToJson()
(o *ChannelUnreadAt) ToJson()
(o *ChannelStats) ToJson()
(o *ChannelView) ToJson()
(o *ChannelViewResponse) ToJson()
(o *ClusterDiscovery) ToJson()
(ci *ClusterInfo) ToJson()
(cs *ClusterStats) ToJson()
(o *Command) ToJson()
CommandListToJson(l []*Command) string
(o *CommandArgs) ToJson()
(cmr *CommandMoveRequest) ToJson()
(o *CommandResponse) ToJson()
(c *Compliance) ToJson()
(c Compliances) ToJson()
(o *Config) ToJson()
EmojiListToJson(emojiList []*Emoji)
```

### CODE REVIEW

File: plugin/helpers_config.go [hunk 1]
```diff
@@ -24,7 +28,17 @@ func (p *HelpersImpl) CheckRequiredServerConfiguration(req *model.Config) (bool,
...
 	mergedCfg := mc.(model.Config)
-	if mergedCfg.ToJson() != cfg.ToJson() {
+	cfgBuf, err := json.Marshal(cfg)
+	if err != nil {
+		return false, fmt.Errorf("failed to marshal config: %v", err)
+	}
+
+	mergedCfgBuf, err := json.Marshal(mergedCfg)
+	if err != nil {
+		return false, fmt.Errorf("failed to marshal merged config: %v", err)
```
Comment (streamer45) [inline plugin/helpers_config.go:38]: Should we use `%w` here or is it the same?

File: testlib/resources.go [hunk 2]
```diff
@@ -206,8 +207,13 @@ func setupConfig(configDir string) error {
 		return errors.Wrapf(err, "failed to create config directory %s", configDir)
 	}
 
+	buf, err := json.Marshal(config)
+	if err != nil {
+		return fmt.Errorf("failed to marshal config: %v", err)
```
Comment (streamer45) [inline testlib/resources.go:212]: Same question.

**Review by streamer45 — APPROVED** [review]
Great work :+1:

File: plugin/helpers_config.go (hunk 1 above)
Comment (agnivade) [reply plugin/helpers_config.go:38]: Don't feel the need to. The error is not a named one, and I don't foresee any scenarios inspecting the inner error.

**Pull request merged**